### PR TITLE
Add workflows for sanity, linters and changelog

### PR DESCRIPTION
**Pull request by alinabuzachis**
Add workflows for sanity, linters and changelog

### CODE REVIEW

File: .github/workflows/sanity.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+              "ansible-version": "stable-2.9"
+            },
+            {
+              "ansible-version": "stable-2.12",
+              "python-version": "3.7"
+            },
+            {
+              "ansible-version": "stable-2.12",
+              "python-version": "3.11"
+            },
```
Comment (gravesm) [inline .github/workflows/sanity.yml:31]: We shouldn't be testing 2.12 at all. It's no longer supported.

File: CI.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+##  cloud.aws_ops Collection
+
+GitHub Actions are used to run the Continuous Integration for redhat-cop/cloud.gcp_ops collection. The workflows used for the CI can be found [here](https://github.com/redhat-cop/cloud.gcp_ops/tree/main/.github/workflows). These workflows include jobs to run the sanity tests, linters and changelog check. The following table lists the python and ansible versions against which these jobs are run.
+
+| Jobs | Description | Python Versions | Ansible Versions |
+| ------ |-------| ------ | -----------|
+| changelog |Checks for the presence of Changelog fragments | 3.9 | devel |
+| Linters | Runs `ansible-lint`, `black` and `flake8` on plugins and tests | 3.8, 3.9(ansible-lint) | devel |
+| Sanity | Runs ansible sanity checks | 3.8, 3.9, 3.10, 3.11 | Stable-2.12, 2.13, 2.14 (not on py 3.11), Stable-2.15+ (not on 3.8) |
```
Comment (gravesm) [inline CI.md:11]: We can drop 2.12 here.

File: requirements.txt
Comment (abikouo) [inline requirements.txt:1]: is this empty file needed ?

File: test-requirements.txt
Comment (abikouo) [inline test-requirements.txt:1]: same here

File: bindep.txt
Comment (abikouo) [inline bindep.txt:1]: this should be removed

File: meta/runtime.yml [hunk 3]
```diff
@@ -1,52 +1,2 @@
...
-# import_redirection:
-#   ansible_collections.ns.col.plugins.module_utils.old_location:
-#     redirect: ansible_collections.ns.col.plugins.module_utils.new_location
-
-# Groups of actions/modules that take a common set of options
-# action_groups:
-#   group_name:
-#     - module1
-#     - module2
+requires_ansible: '>=2.12.0'
```
Comment (gravesm) [inline meta/runtime.yml:2]:
```suggestion
requires_ansible: '>=2.13.0'
```

File: README.md [hunk 4]
```diff
@@ -1,3 +1,95 @@
...
+
+## Included content
+
+Click on the name of a role to view that content's documentation:
+
+<!--start collection content-->
+### Roles
+Name | Description
+--- | ---
+[cloud.gcp_ops.manage_machine_image](https://github.com/ansible-collections/cloud.aws_ops/blob/main/roles/manage_machine_image/README.md)|A role to create / delete a GCP custom machine image.
```
Comment (abikouo) [inline README.md:20]: This should be added in another PR once the role is merged

File: tox.ini [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+skipsdist = True
+
+[testenv]
+deps = -r{toxinidir}/requirements.txt
+       -r{toxinidir}/test-requirements.txt
+install_command = pip install {opts} {packages}
+
+[testenv:black]
+deps =
+  black >= 22.0, < 23.0
```
Comment (gravesm) [inline tox.ini:12]: Why would we not use `>= 23.0, < 24.0`?

**Review by abikouo — APPROVED** [review]

File: .github/workflows/sanity.yml (hunk 1 above)
Comment (alinabuzachis) [reply .github/workflows/sanity.yml]: I assume defaults are picked up from https://github.com/ansible-network/github_actions/blob/main/.github/workflows/sanity.yml#L19, so I entirely removed any 2.12 combination.

File: .config/ansible-lint.yml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+---
+profile: production
+strict: true
+skip_list:
+  - ignore-errors  # Use failed_when and specify error conditions instead of using ignore_errors.
+  - meta-no-info  # meta/main.yml should contain relevant info.
+  - latest[git]  # Result of the command may vary on subsequent runs.
+  - no-handler  # Tasks that run when changed should likely be handlers.
```
[the remark this answers is not in the document]
Comment (alinabuzachis) [reply .config/ansible-lint.yml]: Completely removed for now.

File: README.md (hunk 4 above)
Comment (alinabuzachis) [reply README.md]: Removed!

File: .github/workflows/sanity.yml (hunk 1 above)
Comment (gravesm) [reply .github/workflows/sanity.yml]: Yeah, I believe that should work.

**Review by gravesm — APPROVED** [review]
